src/coreclr/vm/codeman.cpp

```diff
@@ -1525,7 +1525,16 @@ void EEJitManager::SetCpuInfo()
 
     if (((cpuFeatures & ARM64IntrinsicConstants_Sve) != 0) && CLRConfig::GetConfigValue(CLRConfig::EXTERNAL_EnableArm64Sve))
     {
-        CPUCompileFlags.Set(InstructionSet_Sve);
+        int maxVectorLength = (maxVectorTBitWidth >> 3);
+        uint64_t systemVectorTLength = GetSystemVectorLength();
+
+        if (maxVectorLength >= systemVectorTLength)
+        {
+            // Enable SVE only when user specified vector length larger than or equal to the system
+            // vector length. When eabled, SVE would use full vector length available to the process.
+            // For a 256-bit machine, if user provides DOTNET_MaxVectorTBitWidth=128, disable SVE.
```

Comment:
When `DOTNET_MaxVectorTBitWidth` is not specified, it is `0` and in that case, we should use the full vector length, the system offers (as the comment says). In that case, we should set `CPUCompileFlags.Clear(InstructionSet_VectorT256);` on 256-bit machine and `CPUCompileFlags.Clear(InstructionSet_VectorT256);`  and `CPUCompileFlags.Clear(InstructionSet_VectorT512);`  on a 512-bit machine.

Reply:
We wouldn't want to `CPUCompileFlags.Clear`, as that's implicit

Rather instead we would `CPUCompileFlags.Set(InstructionSet_VectorT256);` if the reported SVE length is `256-bits` and `CPUCompileFlags.Set(InstructionSet_VectorT512);` if the reported SVE length is `512-bits`.

They should be off unless otherwise set, but we double check that via a cleanup check anyways (which should be moved to be shared with Arm64): https://github.com/dotnet/runtime/blob/22aa47e02bb28f5fc009f06b7f526ee5aa33dfd7/src/coreclr/vm/codeman.cpp#L1564-L1573

-- I think we also need a TODO here explaining that we're artificially restricting the size to 128-bits for the time being, as the support for larger vector sizes hasn't been plumbed through for Arm64 yet.

Reply:
> We wouldn't want to CPUCompileFlags.Clear, as that's implicit

Yes, that's what I meant...if they are already enabled, then we should disable it or vice-versa. 

> I think we also need a TODO here explaining that we're artificially restricting the size to 128-bits for the time being,

Yes, I think that's what we should do and leave the change of setting `Vector256`, etc. for later PR, when we address some of the things in https://github.com/dotnet/runtime/issues/101477 , specifically `getVectorTByteLength()`.